projects/tracker/SensorTracker.rs

```diff
@@ -0,0 +1,96 @@
+#![no_std]
+#![no_main]
+#![feature(lang_items, start, core, no_std)]
+#![feature(collections, alloc)]
+
+#[macro_use]
+extern crate core;
+extern crate emlib;
+extern crate libc;
+extern crate alloc;
+
+#[macro_use]
+extern crate collections;
+
+use core::prelude::*;
+use core::default::Default;
+
+use emlib::{chip, emu, rtc};
+use emlib::modules::{Usart};
+use emlib::utils::cmdparse::{get_command, Cmd};
+use emlib::stk::io::{PB0, PB1};
+
+use ram_store as store;
+
+mod temperature;
+mod ram_store;
+
+enum State {
+    Connected,
+    Unconnected
+}
+
+static mut MODE: State = State::Unconnected;
+
+#[no_mangle]
+pub extern fn main() {
+
+    chip::init();
+
+    PB0.init(); PB0.on_click(btn0_cb);
+
+    PB1.init(); PB1.on_click(btn1_cb);
+
+    temperature::init(10, false);
+    store::init();
+
+    let mut uart: Usart = Default::default();
+    uart.init_async();
+
+    loop {
+        match unsafe { &MODE } {
+            &State::Connected => match get_command() {
+                Cmd::Read(page) => read(page as usize),
+                _ => ()
+            },
+            _ => emu::enter_em3(true)
+        }
+    }
+}
+
+#[no_mangle]
+#[allow(non_snake_case)]
+pub extern fn RTC_IRQHandler() {
+
+    rtc::int_clear(rtc::RTC_IEN_COMP0);
+    temperature::on_rtc();
```

Comment:
Suggestion: Is the temperature module the only thing using RTC? If so, it is possible can replace the on_rtc() with the actual RTC_IRQHandler.